test/test_schema.py

```diff
@@ -290,3 +291,29 @@ def fn(q: InputC) -> None:
     }
     assert input_schema == expected_input_schema
     assert is_validate_dict(input_schema)
+
+
+def test_schema_to_base_model():
+    def fn(
+        a: int,
+        b: Union[float, int] = 4,
+        c: Optional[str] = "my_string",
+        d: bool = False,
+        e: dict[str, Any] = None,
+        f: list[float] = None,
+    ) -> None:
+        pass
+
+    class ExpectedInputModel(BaseModel):
+        a: int
+        b: Union[float, int] = 4
+        c: Optional[str] = "my_string"
+        d: bool = False
+        e: dict = None
```

Comment:
The main issue is that type subscription is lost. This creates discrepancy against the output of `/schema` endpoint, which shows full typing.

My proposal is to accept this solution and create a Github issue (or any other trackable item) which will be addressed later.

Reply:
> type subscription is lost

What do you mean by this?

Reply:
For example `e: dict[str, Any]` in `fn` is transformed into `e: dict` in `ExpectedInputModel`. Typing in brackets for collection types is omitted.